cpp/ql/src/Security/CWE/CWE-428/UnsafeCreateProcessCall.ql

```diff
@@ -0,0 +1,135 @@
+/**
+ * @name NULL application name with an unquoted path in call to CreateProcess
+ * @description Calling a function of the CreatePorcess* family of functions, which may result in a security vulnerability if the path contains spaces.
+ * @id cpp/unsafe-create-process-call
+ * @kind problem
+ * @problem.severity error
+ * @precision medium
+ * @tags security
+ *       external/cwe/cwe-428
+ *       external/microsoft/C6277
+ */
+ 
+import cpp
+import semmle.code.cpp.dataflow.DataFlow
+import semmle.code.cpp.dataflow.DataFlow2
+
+/**
+ * A function call to CreateProcess (either wide-char or single byte string versions)
+ */
+class CreateProcessFunctionCall extends FunctionCall {
+  CreateProcessFunctionCall() {
+    (
+      this.getTarget().hasGlobalName("CreateProcessA") or
+      this.getTarget().hasGlobalName("CreateProcessW") or
+      this.getTarget().hasGlobalName("CreateProcessWithTokenW") or
+      this.getTarget().hasGlobalName("CreateProcessWithLogonW") or
+      this.getTarget().hasGlobalName("CreateProcessAsUserA") or
+      this.getTarget().hasGlobalName("CreateProcessAsUserW") 
+    )
+  }
+  
+  int getApplicationNameArgumentId() {
+    if(
+      this.getTarget().hasGlobalName("CreateProcessA") or
+      this.getTarget().hasGlobalName("CreateProcessW") 
+    ) then ( result = 0 )
+    else if (
+      this.getTarget().hasGlobalName("CreateProcessWithTokenW") 
+    ) then ( result = 2 )
+    else if (
+      this.getTarget().hasGlobalName("CreateProcessWithLogonW") 
+    ) then ( result = 4 )
+    else if(
+      this.getTarget().hasGlobalName("CreateProcessAsUserA") or
+      this.getTarget().hasGlobalName("CreateProcessAsUserW") 
+    ) then ( result = 1 )
+    else (result = -1 )
```

Comment:
The `-1` cases in this predicate and the one below could be omitted.  It's perfectly fine for a predicate to have no result, and either way a use of `FunctionCall.getArgument(thisPredicate())` would have no result.

Reply:
Got it. I will fix it & submit the change

Reply:
The best thing to do is to rewrite those predicates using `and` and `or`:
```
(
  this.getTarget().hasGlobalName("CreateProcessA") or
  this.getTarget().hasGlobalName("CreateProcessW")
) and
result = 0
or
...
```

Reply:
Never mind, with Dave's recommendation below, I don't need the IF anymore. Thanks